===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(index_build:stack->elem)) at 51 (line 374, column 3),
        Insert: if (true) free(*(index_build:stack)) at 52 (line 374, column 3) }
===== Actual patch in diff format =====
@@ -371,7 +371,12 @@ index_build (p11_index *index,
 
 		/* Make a shallow copy of the combined attributes for validation */
 		built = calloc (nmerge + nattrs + nextra + 1, sizeof (CK_ATTRIBUTE));
-		return_val_if_fail (built != NULL, CKR_GENERAL_ERROR);
+		if (built != NULL) { } else {
+			p11_debug_precond ("p11-kit: '%s' not true at %s\n", "built != NULL", __func__);
+			free(stack->elem);
+			free(stack);
+			return CKR_GENERAL_ERROR;
+		}
 
 		count = nmerge;
 		memcpy (built, merge, sizeof (CK_ATTRIBUTE) * nmerge);
